### ISSUE TITLE
The entity inside a Prefab is not deleted on the first attempt

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#5078]
**Describe the bug**
The entities inside a Prefab require two presses of the _Delete_ button to be successfully deleted. This occurs when deleting the entity in all possible ways (i.e. by pressing the _DEL_ key on the keyboard, as well as by pressing the _Delete_ option in the entity context menu).

Additionally, when the User attempts to delete an entity for the first time, the following warnings are printed in the Console (please note that the warnings and error below do not always reproduce):
`[Warning] (Prefabs) - Prefab - EntityIdMapper: Entity with Id [9893601651242557910] has no registered owning instance`

After the second attempt, the following warning and error are also printed in the Console:
```
[Warning] (Prefab Serialization) - The "remove" operation failed to remove member 'Entity_[568973499592]' from object at path '/Entities/Entity_[568973499592]'.
 Reason: The operation has halted during a merge operation for '/0' because a field or element has an invalid value.
[Error] (Prefab) - Some of the patches were not successfully applied.
```

Please refer to the attached video for more details.

**Steps to reproduce**
1. Open any level.
2. Create a new entity and make a Prefab out of it.
3. Double click the Prefab to enter its Focus Mode.
4. Select any entity inside a Prefab.
5. Press _DEL_ on the keyboard.

**Expected behavior**
Entity is successfully deleted.

**Actual behavior**
Entity is not deleted, warnings appear in the Console.

**Assets required**
N/A

**Video**

https://user-images.githubusercontent.com/86953108/139243061-6d3c0118-3fd3-40cf-acf4-2715bf2434e1.mp4

**Found in Branch**
Stabilization_2110 (caab78d)

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version: 10
 - CPU: AMD Ryzen 5 3600
 - GPU: Nvidia RTX 2060 SUPER
 - Memory: 16GB

**Comment by pollend** [comment]
so when you create a prefab it seems like an instance is never associated with the entities inside of a prefab. I need to look at this much closer but I think I have a basic clue.

**Comment by srikappa-amzn** [comment]
@pollend , when you create a prefab, the entities are in fact associated with the instance. They are the 'm_entities' under AzToolsFramework::Prefab::Instance. 
This became an issue due to a couple of changes we made around prefab propagation model. I'm working on reverting those changes so that we can get delete working as usual. 
Just wanted to let you know so that we can avoid duplicate efforts on this.

**Comment by srikappa-amzn** [comment]
This is now fixed with PR https://github.com/o3de/o3de/pull/5174  in branch https://github.com/o3de/o3de/tree/stabilization/2110

**Comment by LB-TomaszSerwanski** [comment]
The issue verified as Fixed in Stabilization/2110 (d598a7c) but it is still active in Development(cdf5a05).

**Comment by srikappa-amzn** [comment]
The integration from stabilization/2110 to development happens every 2 weeks. Looks like we just didn't missed the cut this time. This should be fixed in development too when the next merge happens in about 2 weeks.
